src/references.rs

```diff
@@ -17,3 +24,168 @@ pub fn null_mut<'a, T: 'static>() -> &'a mut T {
 pub fn not_alloc<'a, T: 'static>() -> &'a mut T {
 	crate::transmute(usize::MAX)
 }
+
+/// Easily dereferencable raw pointer. Can be freely moved or copied. Do you really
+/// desire to have such power? If so, do as you wish. You've been warned.
+///
+/// `Ref<T>` is guaranteed to be the same size as `usize`
+#[derive(Copy, PartialEq, Eq, PartialOrd, Ord)]
+#[repr(C)]
+pub struct Ref<T>(usize, PhantomData<Arc<T>>)
+where
+	T: Sized;
+impl<T> Clone for Ref<T> {
+	fn clone(&self) -> Self {
+		Self(self.0, PhantomData)
+	}
+}
+impl<T> Ref<T> {
+	pub fn new(value: T) -> Self {
+		Self(Box::leak(Box::new(value)) as *const _ as usize, PhantomData)
+	}
+
+	pub fn free(self) {
+		let boxx: Box<T> = crate::transmute(self);
+		drop(boxx);
+	}
+
+	pub fn addr(&self) -> usize {
+		self.0
+	}
+
+	pub fn as_ptr(&self) -> *const T {
+		crate::transmute(self.0)
+	}
+
+	pub fn as_ptr_mut(&mut self) -> *mut T {
+		crate::transmute(self.0)
+	}
+
+	pub fn from_ptr(ptr: *const T) -> Self {
+		Self(ptr as usize, PhantomData)
+	}
+
+	pub fn from_ptr_mut(ptr: *mut T) -> Self {
+		Self(ptr as usize, PhantomData)
+	}
+
+	pub fn into_box(&self) -> Box<T> {
```

Comment:
maybe `impl Into<Box<T>>` instead of a method? Honestly not sure if that's any better or not. Could also do `Into<*const T>`/`Into<*mut T>`. Idrk how I feel about that, just something that popped in my head.

Reply:
There were attempts
![image](https://github.com/Speykious/cve-rs/assets/42136194/168d56ee-0d80-4d5e-86c8-83c10f90b894)

Reply:
It seems like implementing `From<T>` implicitly implements `Into<T>`. I have no idea how that works, but ig it does 🤷🏻

Reply:
If you have 2 structs `Foo` and `Bar`, and `Bar` implements `From<Foo>`, then `Foo` implements `Into<Bar>` because it can call `Bar::from(Foo)`.

Oh well, the method's fine anyways. Man we really need specificity to get stabilised :/